### PR TITLE
feat(integrations): Bitbucket issue and repo search

### PR DESCRIPTION
**Pull request by MeredithAnya**
* autocomplete search for issues for bitbucket
* autocomplete search for repos (both in issues and repo dropdown)

### CODE REVIEW

File: src/sentry/integrations/bitbucket/issues.py [hunk 1]
```diff
@@ -70,6 +71,11 @@ def get_create_issue_config(self, group, **kwargs):
     def get_link_issue_config(self, group, **kwargs):
         repo_choices, default_repo, issues = self.get_repo_choices(**kwargs)
```
Comment (EvanPurkhiser) [inline src/sentry/integrations/bitbucket/issues.py]: I'm changing this in a separate PR, we don't need `get_repo_choices` also returning issues (which we're now not even using anymore)

**Review by mnoble — CHANGES_REQUESTED** [review]
Looks good overall – should add some tests. At the least for the endpoint.

File: src/sentry/integrations/bitbucket/issues.py [hunk 2]
```diff
@@ -36,19 +37,25 @@ def get_repo_choices(self, **kwargs):
...
+
         return [
             {
                 'name': 'repo',
                 'label': 'Bitbucket Repository',
                 'type': 'select',
                 'default': default_repo,
-                'choices': repo_choices,
+                'defaultLabel': default_repo,
+                'url': autocomplete_url,
```
[the remark this answers is not in the document]
Comment (MeredithAnya) [reply src/sentry/integrations/bitbucket/issues.py]: > Does it pass `name` as a query param somehow? 

Yeah - in the search endpoint it's the field so we know whether we are searching 'repo' vs. 'externalIssue' vs. etc

**Review by mnoble — APPROVED** [review]
👍

**Pull request merged**